src/freedv_interface.cpp

```diff
@@ -93,9 +121,21 @@ void FreeDVInterface::start(int txMode, int fifoSizeMs, bool singleRxThread, boo
     for (auto& mode : enabledModes_)
     {
         struct freedv* dv = nullptr;
-        if ((mode == FREEDV_MODE_700D) || (mode == FREEDV_MODE_700E) || (mode == FREEDV_MODE_2020)) {
+        if ((mode == FREEDV_MODE_700D) || (mode == FREEDV_MODE_700E) || (mode == FREEDV_MODE_2020) ||
+            (mode == FREEDV_MODE_2020A) || (mode == FREEDV_MODE_2020B)) {
             // 700 has some init time stuff so treat it special
             struct freedv_advanced adv;
+            
+            if (mode == FREEDV_MODE_2020A || mode == FREEDV_MODE_2020B)
+            {
+                // Set index optimized VQ to 1 (required for 2020A/B).
+                adv.lpcnet_vq_type = 1;
+            }
+            else if (mode == FREEDV_MODE_2020)
+            {
+                adv.lpcnet_vq_type = 0;
+            }
+            
             dv = freedv_open_advanced(mode, &adv);
```

Comment:
Suggest we just use `freedv_open()` for all modes - that will set up index optimisation correctly.  FYI using **0** is a predictive quantiser, where we quantise the differences from the last frame.  Efficient, but bit errors tend to propagate so it degrades badly in the sort of error prone channels we use for HF digital voice. Vanilla 2020 currently uses **1**, but no need to expose all this at the `freedv-gui` level.

Reply:
Updated accordingly. 👍